operator/cmd/mesh/upgrade.go

```diff
@@ -164,6 +166,23 @@ func upgrade(rootArgs *rootArgs, args *upgradeArgs, l clog.Logger) (err error) {
 		return fmt.Errorf("failed to read the current Istio version, error: %v", err)
 	}
 
+	// Upgrade won't work, if the previous version of Istio is installed with revision
+	// Ref: https://github.com/istio/istio/issues/28566
+	pods, err := kubeClient.PodsForSelector(istioNamespace, "app=istiod")
+	if err != nil {
+		return fmt.Errorf("failed to fetch istiod pod, error: %v", err)
```

Comment:
-> "failed to fetch the existing istiod pod for checking revision, error: %v"

Reply:
Done.